mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -273,11 +273,16 @@ public MessageManager getMailbox(MailboxPath mailboxPath, MailboxSession session
             LOGGER.info("Mailbox '{}' not found.", mailboxPath);
             throw new MailboxNotFoundException(mailboxPath);
 
-        } else {
-            LOGGER.debug("Loaded mailbox {}", mailboxPath);
+        }
 
-            return createMessageManager(mailboxRow, session);
+        if (!assertUserHasAccessTo(mailboxRow, session)) {
+            LOGGER.info("Mailbox '{}' does not belong to user '{}' but to '{}'", mailboxPath, session.getUser(), mailboxRow.getUser());
+            throw new MailboxNotFoundException(mailboxPath);
         }
+
+        LOGGER.debug("Loaded mailbox {}", mailboxPath);
```

Comment:
It's not really loaded, is it?

Reply:
That log was already there, I just moved it (it's the same log for the getMailbox by id). I thought that was acceptable, as we return a MessageManager with which allows us to perform operations on that mailbox, thus "loaded". Do you have a better log to propose?